src/algebraic_geometry/morphisms/basic.lean

```diff
@@ -0,0 +1,165 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import algebraic_geometry.AffineScheme
+import algebraic_geometry.pullbacks
+
+/-!
+# Properties of morphisms
+
+We provide the basic framework for talking about properties of morphisms.
+The following meta-properties are defined
+
+* `respects_iso`: `P` respects isomorphisms if `P f → P (e ≫ f)` and `P f → P (f ≫ e)`, where
+  `e` is an isomorphism.
+* `stable_under_composition`: `P` is stable under composition if `P f → P g → P (f ≫ g)`.
+* `stable_under_base_change`: `P` is stable under base change if `P (Y ⟶ S) → P (X ×[S] Y ⟶ X)`.
+
+-/
+
+universe u
+
+open topological_space category_theory category_theory.limits opposite
+
+noncomputable theory
+
+namespace algebraic_geometry
+
+/-- A `morphism_property` is a class of morphisms between schemes. -/
+def morphism_property := ∀ ⦃X Y : Scheme⦄ (f : X ⟶ Y), Prop
+
+/-- An `affine_target_morphism_property` is a class of morphisms from an arbitrary scheme into an
+affine scheme. -/
+def affine_target_morphism_property := ∀ ⦃X Y : Scheme⦄ (f : X ⟶ Y) [is_affine Y], Prop
+
+/-- `is_iso` as an `morphism_property`. -/
+protected def is_iso : morphism_property := @is_iso Scheme
+
+/-- `is_iso` as an `morphism_property`. -/
+protected def affine_target_is_iso : affine_target_morphism_property :=
+λ X Y f H, is_iso f
+
+instance : lattice morphism_property :=
```

Comment:
Can't you just use [pi.complete_lattice](https://leanprover-community.github.io/mathlib_docs/order/complete_lattice.html#pi.complete_lattice) and [Prop.complete_lattice](https://leanprover-community.github.io/mathlib_docs/order/complete_lattice.html#Prop.complete_lattice)? It probably works if you just add `@[derive complete_lattice]` to `def morphism_property`.